receiver/datadogreceiver/translator.go

```diff
@@ -71,21 +71,30 @@ func toTraces(payload *pb.TracerPayload, req *http.Request) ptrace.Traces {
 				}
 			}
 
-			switch span.Type {
-			case "web":
-			case "server":
-				newSpan.SetKind(ptrace.SpanKindServer)
-			case "client":
-				newSpan.SetKind(ptrace.SpanKindClient)
-			case "producer":
-				newSpan.SetKind(ptrace.SpanKindProducer)
-			case "consumer":
-				newSpan.SetKind(ptrace.SpanKindConsumer)
-			case "internal":
-				newSpan.SetKind(ptrace.SpanKindInternal)
-			case "custom":
-			default:
-				newSpan.SetKind(ptrace.SpanKindUnspecified)
+			if span.Meta["span.kind"] != "" {
```

Comment:
Couldn't you do something like

 str, ok := span.Meta["span.kind"]; ok {
 // CODE HERE
}

This is more the go way, no?

Reply:
Yes, it's more go way. But we need to double check since we cannot garantee `span.Meta["span.kind"]` is not empty. It is sent over network. `if kind, ok := span.Meta["span.kind"]; ok && kind != "" {` is more verbose.

Reply:
Meta will exist there always, this happens after the tags have been ported